src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+    end
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.upper - f.constant
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.lower - f.constant
+end
+
+
+function loadObj(cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction)
+    for term in f.terms
+        cbcModelFormat.obj[mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+end
+
+
+```
+    Create inner model cbcOptimizer based on abstract model userOptimizer provided by user.
+    Fill the object of type CbcModelFormat:
+        constraint_matrix::AbstractMatrix,
+        col_lb::VecOrNothing,
+        col_ub::VecOrNothing,
+        obj::VecOrNothing,
+        row_lb::VecOrNothing,
+        row_ub::VecOrNothing,
+    These are needed by function loadProblem of CbcCInterface.
+
+```
+function MOI.copy!(cbcOptimizer::CbcOptimizer,
+    userOptimizer::MOI.ModelLike; copynames=false)
+
+    mapping = MOIU.IndexMap()
+
+    nbCols = MOI.get(userOptimizer, MOI.NumberOfVariables())
+    varIndex = MOI.get(userOptimizer, MOI.ListOfVariableIndices())
+    for i in 1:nbCols
+        mapping.varmap[varIndex[i]] = MOI.VariableIndex(i)
+    end
+
+    zeroOneIndices = Vector{Int}(0)
+    integerIndices = Vector{Int}(0)
+    listOfConstraints = MOI.get(userOptimizer, MOI.ListOfConstraints())
+    nbRows = 0;
+    for (F,S) in listOfConstraints
+        (F,S) in SUPPORTED_CONSTRAINTS || return MOI.CopyResult(MOI.CopyUnsupportedConstraint, "Cbc MOI Interface does not support constraints of type " * (F,S) * ".", nothing)
+
+        ci = MOI.get(userOptimizer, MOI.ListOfConstraintIndices{F,S}())
+
+        if (F,S) in INTEGRALITY_CONSTRAINTS
+            for i in 1:length(ci)
+                f = MOI.get(userOptimizer, MOI.ConstraintFunction(), ci[i])
+                S == MOI.ZeroOne && push!(zeroOneIndices, mapping.varmap[f.variable].value)
+                S == MOI.Integer && push!(integerIndices, mapping.varmap[f.variable].value)
+            end
+        end
+
+        ## Update conmap for (F,S) constraints
+        F == MOI.SingleVariable && continue
+        for i in 1:length(ci)
+            mapping.conmap[ci[i]] = MOI.ConstraintIndex{F,S}(nbRows + i)
+        end
+        nbRows += MOI.get(userOptimizer, MOI.NumberOfConstraints{F,S}())
+
+    end
+
+    cbcModelFormat = CbcModelFormat(nbRows, nbCols)
+
+    for (F,S) in listOfConstraints
+        (F,S) in INTEGRALITY_CONSTRAINTS && continue
+        for ci in MOI.get(userOptimizer, MOI.ListOfConstraintIndices{F,S}())
+            f = MOI.get(userOptimizer, MOI.ConstraintFunction(), ci)
+            s = MOI.get(userOptimizer,  MOI.ConstraintSet(), ci)
+            loadConstraint(ci, cbcModelFormat, mapping, f, s)
+        end
+    end
+
+    ## Add bounds to binary columns if it makes sense
+    for colIdx in zeroOneIndices
+        if cbcModelFormat.col_lb[colIdx] < 0.0
+            cbcModelFormat.col_lb[colIdx] = 0.0
```

Comment:
Does Cbc require this? Otherwise if the user deletes a Binary constraint later the bounds will be off

Reply:
Cbc.jl only implements `setInteger`, so a constraint with a binary `MOI.Set` may not be correctly implemented without this bound verifications. And since cbc does not support problem modification, the user must go through `copy!`, hence it is expected that he provides a "correct" problem. 

Or should we just say that cbc does not support binary variables?